feat: add multi-facility tutorial (#1468)
medic/cht-docs: content/en/apps/tutorials/multi-facility-users.md

---
title: "Setting up Multi-facility users"
linkTitle: Multi Facility Users
weight: 6
description: >
  Creating and assigning users to multiple places in the CHT UI
relatedContent: >
  apps/concepts/users
  apps/reference/app-settings
  apps/tutorials/contact-and-users-1
---

{{% pageinfo %}}
This tutorial will take you through how to create and assign users to multiple places in the CHT UI, including;

- Create contacts and their associated users
- Create places and assign contacts to those places
- Assigning users to multiple places

> App settings allow you to both persist information that is critical to the application outside the code, and to create profiles that store the preferences for project deployments.

Reviewer: Do you mean the admin app? 

```suggestion
The [App Management]({{< relref "apps/features/admin" >}}) allows you to both persist information that is critical to the application outside the code, and to create profiles that store the preferences for project deployments.
```

Or the actual app-settings file? 

```suggestion
The [CHT application settings]({{< relref "apps/tutorials/application-settings" >}}) allows you to both persist information that is critical to the application outside the code, and to create profiles that store the preferences for project deployments.
```

Author: I meant the app-settings file.

For the second Health Facility, we will skip creating or assigning a primary contact and focus on creating the second Health Facility.

<br clear="all">

{{< figure src="new-facility/enter-second-facility-name.png" link="new-facility/enter-second-facility-name.png" class="right col-6 col-lg-8" >}}

Enter the details of the Health Facility and submit the form.

<br clear="all">

{{< figure src="new-facility/multiple-facilities.png" link="new-facility/multiple-facilities.png" class="right col-6 col-lg-8" >}}

You should see the two Health Facilities appear on the left-hand side.

<br clear="all">

### 4. Create the CHW Supervisor User

Let's create a CHW Supervisor user who's linked to the CHW Supervisor contact we created earlier.

<br clear="all">

{{< figure src="new-chw-supervisor/app-settings.png" link="new-chw-supervisor/app-settings.png" class="right col-6 col-lg-8" >}}

Go to the **hamburger menu** and select **App Settings**.

<br clear="all">

{{< figure src="new-chw-supervisor/add-user.png" link="new-chw-supervisor/add-user.png" class="right col-6 col-lg-8" >}}

> When you are on the **App Settings** page, select **Users** on the left-hand side and then select **Add User** on the right-hand side.

Reviewer: ```suggestion
When you are on the **App Management** page, select **Users** on the left-hand side and then select **Add User** on the right-hand side.
```
In `default` config and [cht docs](https://docs.communityhealthtoolkit.org/apps/features/admin/), it's called different: 

<img width="793" alt="Screenshot 2024-07-29 at 3 18 12 PM" src="https://github.com/user-attachments/assets/7fb2bcce-f52e-4c8c-9920-22c39cfa7e54">

---
title: "Setting up Multi-facility users"
linkTitle: Multi Facility Users
weight: 6
description: >
  Creating and assigning users to multiple places in the CHT UI
relatedContent: >
  apps/concepts/users
  apps/reference/app-settings
  apps/tutorials/contact-and-users-1
---

> {{% pageinfo %}}

Reviewer 2: @Benmuiruri , mention that this is only available from 4.9

This tutorial will take you through how to create and assign users to multiple places in the CHT UI, including;

- Create contacts and their associated users
- Create places and assign contacts to those places
- Assigning users to multiple places

App settings allow you to both persist information that is critical to the application outside the code, and to create profiles that store the preferences for project deployments.
{{% /pageinfo %}}

## Brief Overview of Key Concepts

*[Permissions]({{< relref "apps/concepts/users#permissions" >}})* are settings that control access to specific app features and functionality.

*[Roles]({{< relref "apps/concepts/users#roles" >}})* define permissions for users to access a group of app features and functionality.

*[Contacts]({{< relref "apps/features/contacts" >}})*  are people or places that are created in the CHT application.

*[People]({{< relref "apps/features/contacts" >}})* are both patients in the system and users of the system, such as CHWs or Nurses.

*[Places]({{< relref "apps/features/contacts" >}})* represent either an actual physical location such as a health facility, clinic, or a grouping such as a household or CHW Area.

> ## Required Resources

Reviewer 2: ```suggestion
## Pre-requisites
```

- You should be familiar with contact and user management. View detailed tutorial on [How to create contacts and their associated users]({{% ref "apps/tutorials/contact-and-users-1" %}}) 
- You should be familiar with managing CHT application settings. Read [How to manage CHT application settings]({{% ref "apps/tutorials/application-settings" %}}) 

## Implementation Steps

In this tutorial, you will work with the default contact forms and the default hierarchy.

The default `app_settings.json` in the CHT found in the `/config/default` folder has an existing `chw_supervisor` role and the `"can_have_multiple_places": []` permission. 

> While logged in as an admin user, you will create two Health Facilities, CHW Supervisor, and CHW Area. You will then create the user for the CHW Supervisor so that they can log in see their facilities, and the CHW Areas they supervise.

Reviewer 2: ```suggestion
While logged in as an admin user, you will create two Health Facilities, CHW Supervisor, and CHW Area. You will then create the user for the CHW Supervisor so that they can log in and see the facilities, and the CHW Areas they supervise.
```
